Improve the description and names of the openId features (#16697)
OrchardCMS/OrchardCore: src/docs/reference/modules/OpenId/README.md

## OpenID Connect Module

> `OrchardCore.OpenId` provides robust OpenID Connect functionality, enabling Orchard Core to act as an OpenID Connect Server and Client. The following features are available:

Reviewer: ```suggestion
`OrchardCore.OpenId` provides robust OpenID Connect functionality, enabling Orchard Core to act as an OpenID Connect server and client. The following features are available:
```

> - **Core OpenID Connect Services**

Reviewer: Should we rename that to match the new name of the feature?

```suggestion
- **OpenID Connect Core Services**
```

- **OpenID Connect Authorization Server**
- **OpenID Connect Management UI**
- **OpenID Connect Authorization Server**
- **OpenID Connect Client Integration**

## Core OpenID Connect Services

This feature provides the essential services that underpin all other OpenID Connect features within Orchard Core. It includes fundamental components needed for secure communication, token handling, and user authentication.

## OpenID Connect Management UI

Adds a management interface to the Orchard Core admin dashboard, enabling users to manage OpenID Connect applications, define and modify scopes, and configure application permissions through a user-friendly UI.

## OpenID Connect Authorization Server

> Allows Orchard Core to function as an OpenID Connect Authorization Server, also referred to as an Identity Provider (IdP). This feature enables authentication and the issuance of tokens, conforming to OpenID Connect and OAuth 2.0 standards.

Reviewer: ```suggestion
Allows Orchard Core to function as an OpenID Connect authorization server, also referred to as an identity provider (IdP). This feature enables authentication and the issuance of tokens, conforming to OpenID Connect and OAuth 2.0 standards.
```

Key points:
- Orchard Core can serve as a centralized identity provider, allowing external applications to authenticate users and manage access control.
- Powered by the [`OpenIddict`](https://github.com/openiddict/openiddict-core) library, this feature supports token-based authentication without requiring an external identity provider.

> - The Authorization Server maintains a private JWT/validation handler for the `/connect/userinfo` API endpoint, so you don't need to enable the Token Validation feature for the current tenant.

Reviewer: Not specific to your PR since that text was already there but maybe something like that should be clearer/less specific?

```suggestion
- The authorization server takes care of validating the access tokens received by the `/connect/userinfo` API endpoint, so you don't need to enable the token validation feature for the current tenant.
```

- To validate issued tokens, ensure the 'OpenID Connect Token Validation' feature is activated.

Supported flows include:
- [Authorization Code Flow](http://openid.net/specs/openid-connect-core-1_0.html)
- [Implicit Flow](http://openid.net/specs/openid-connect-core-1_0.html)
- [Hybrid Flow](http://openid.net/specs/openid-connect-core-1_0.html)
- [Client Credentials Grant](https://tools.ietf.org/html/rfc6749)
- [Resource Owner Password Grant](https://tools.ietf.org/html/rfc6749)

## OpenID Connect Token Validation

> This feature is responsible for validating tokens issued either by Orchard Core's own OpenID Connect Authorization Server or by other trusted servers. It supports JSON Web Tokens (JWT) and OpenID Connect discovery, ensuring secure and reliable token validation across distributed applications.

Reviewer: ```suggestion
This feature is responsible for validating tokens issued either by Orchard Core's own OpenID Connect authorization server or by other trusted servers. It supports JSON Web Tokens (JWT) and OpenID Connect discovery, ensuring secure and reliable token validation across distributed applications.
```
